blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DAbstractMetricsRW.java

```diff
@@ -0,0 +1,324 @@
+    protected final DBasicNumericIO basicIO = new DBasicNumericIO();
+    protected final DSimpleNumberIO simpleIO = new DSimpleNumberIO();
+
+    // a map of RollupType to its IO class that knows
+    // how to read/write that particular type of rollup
+    protected final Map<RollupType, DAbstractMetricIO> rollupTypeToIO =
+            new HashMap<RollupType, DAbstractMetricIO>() {{
+                put(RollupType.BF_BASIC, basicIO );
+                put(RollupType.COUNTER, counterIO);
+                put(RollupType.ENUM, enumIO);
+                put(RollupType.GAUGE, gaugeIO);
+                put(RollupType.SET, setIO);
+                put(RollupType.TIMER, timerIO);
+            }};
+
+
+
+    /**
+     * This method inserts a collection of metrics in the
+     * {@link com.rackspacecloud.blueflood.service.SingleRollupWriteContext}
+     * objects to the appropriate Cassandra column family
+     *
+     * @param writeContexts
+     * @throws IOException
+     */
+    @Override
+    public void insertRollups(List<SingleRollupWriteContext> writeContexts) throws IOException {
+
+        if (writeContexts.size() == 0) {
+            return;
+        }
+
+        Map<ResultSetFuture, SingleRollupWriteContext> futureLocatorMap = new HashMap<ResultSetFuture, SingleRollupWriteContext>();
+        Timer.Context ctx = Instrumentation.getWriteTimerContext( writeContexts.get( 0 ).getDestinationCF().getName() );
+        try {
+            for (SingleRollupWriteContext writeContext : writeContexts) {
+                Rollup rollup = writeContext.getRollup();
+                Locator locator = writeContext.getLocator();
+                Granularity granularity = writeContext.getGranularity();
+                int ttl = getTtl(locator, rollup.getRollupType(), granularity);
+
+                // lookup the right writer
+                RollupType rollupType = writeContext.getRollup().getRollupType();
+                DAbstractMetricIO io = rollupTypeToIO.get(rollupType);
+                if ( io == null ) {
+                    throw new InvalidDataException(
+                            String.format("insertRollups(locator=%s, granularity=%s): unsupported rollupType=%s",
+                                    locator, granularity, rollupType.name()));
+                }
+
+                ResultSetFuture future = io.putAsync(locator, writeContext.getTimestamp(), rollup, writeContext.getGranularity(), ttl);
+                futureLocatorMap.put(future, writeContext);
+            }
+
+            for (ResultSetFuture future : futureLocatorMap.keySet()) {
+                try {
+                    future.getUninterruptibly().all();
+                } catch (Exception ex) {
+                    SingleRollupWriteContext writeContext = futureLocatorMap.get(future);
+                    LOG.error(String.format("error writing to locator %s, granularity %s", writeContext.getLocator(), writeContext.getGranularity()), ex);
```

Comment:
Yep, that's what we for all the metrics stuff.  We return with incomplete data and log the error.